pkg/controllerutils/eventhandler.go

```diff
@@ -36,38 +47,96 @@ func EnqueueCreateEventsOncePer24hDuration(clock clock.Clock) handler.Funcs {
 			if evt.Object == nil {
 				return
 			}
-			q.AddAfter(reconcile.Request{NamespacedName: types.NamespacedName{
-				Name:      evt.Object.GetName(),
-				Namespace: evt.Object.GetNamespace(),
-			}}, getDuration(evt.Object, clock))
+			q.AddAfter(reconcileRequest(evt.Object), getDuration(evt.Object, clock, nil))
 		},
 		UpdateFunc: func(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
 			if evt.ObjectNew == nil {
 				return
 			}
-			q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
-				Name:      evt.ObjectNew.GetName(),
-				Namespace: evt.ObjectNew.GetNamespace(),
-			}})
+			q.Add(reconcileRequest(evt.ObjectNew))
+		},
+		DeleteFunc: func(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+			if evt.Object == nil {
+				return
+			}
+			q.Add(reconcileRequest(evt.Object))
+		},
+	}
+}
+
+// EnqueueWithJitterDelay returns handler.Funcs which enqueues the object with a random Jitter duration when the JitterUpdate
+// is enabled in ManagedSeed controller configuration.
+// All other events are normally enqueued.
+func EnqueueWithJitterDelay(cfg config.ManagedSeedControllerConfiguration) handler.Funcs {
+	return handler.Funcs{
+		CreateFunc: func(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+			managedSeed, ok := evt.Object.(*seedmanagementv1alpha1.ManagedSeed)
```

Comment:
Given that this handler func is specific to `ManagedSeed`, it does not seem to belong into this file which typically contains general-purpose funcs.